### PR TITLE
pass polyfillModuleNames into packager

### PR DESCRIPTION
**Pull request by eacaps**
## Motivation

After examining how React Native sets up `process.env.NODE_ENV` using `global.__DEV__` from `prelude_dev.js` or `prelude.js` by treating them like polyfills I decided to use the same approach for environment variables. I setup my own rn-project.config.js file like so: 

```
const blacklist = require('react-native/packager/blacklist');
const pathJoin = require('path').join;

module.exports = {
  getBlacklistRE: function() {
    return blacklist([/build\/.*/, /app\/assets\/webpack.*/]);
  },
  polyfillModuleNames: [pathJoin(__dirname, 'globals.js')]
};
```

I ran the packaging server using: 
`react-native start --config=config/react-native/rn-project.config.js --reset-cache`

I expected my polyfillModuleNames to be passed into the Packager properly and be handled the same way the built-in polyfills worked. Unfortunately I noticed the Packager wasn't actually getting `opt.polyfillModuleNames`. Digging into the code a bit, it seems the local-cli wasn't passing the polyfillModuleNames from the config.

## Test Plan (required)

There are no specs for runServer.js but this change can be tested by using a config that contains polyfillModuleNames. Sample config and run command provided above simple `global.js` provided below:

```
global.process = global.process ? global.process : {};
global.process.env = global.process.env ? global.process.env : {};
global.process.env['PROJECT_ENV'] = 'staging';
```

### CODE REVIEW

**Comment by eacaps** [comment]
Could I get some feedback on this? It seems like it was just an oversight not to pass opts.polyfillModuleNames into the packager since the packager is already looking for the option itself

**Comment by jeanlauliac** [comment]
I'll look tomorrow

**Comment by eacaps** [comment]
@jeanlauliac did you get a chance to look over this? I could write up a simple test app if you want to see an example

**Comment by eacaps** [comment]
That makes sense to allow more flexibility for people writing their own configs. Here's what my updated config would look like:

```
const blacklist = require('react-native/packager/blacklist');
const pathJoin = require('path').join;

module.exports = {
  getBlacklistRE: function() {
    return blacklist([/build\/.*/, /app\/assets\/webpack.*/]);
  },
  polyfillModuleNames: funtion() {
    return [pathJoin(__dirname, 'globals.js')]
  }
};
```

Making and testing out the changes now.

File: local-cli/server/runServer.js [hunk 1]
```diff
@@ -84,6 +84,9 @@ function getPackagerServer(args, config) {
     args.transformer ? path.resolve(args.transformer) :
     typeof config.getTransformModulePath === 'function' ? config.getTransformModulePath() :
     undefined;
+  const polyfillModuleNames =
+    typeof config.getPolyfillModuleNames === 'function' ? config.getPolyfillModuleNames() :
```
Comment (jeanlauliac) [inline local-cli/server/runServer.js]: Could you make it non-nullable instead? That's the new pattern we tend to stick with for new fields, by providing a default value that is an empty function in the `defaultConfig` object. You can take example on the way we added the new configuration function `getSourceExts`: https://github.com/eacaps/react-native/commit/4a86f93982ede66a993a72e31cadb85d9bd4b3a3
(without going as far as to add it as a command line argument, that I don't think is necessary in that case)

File: local-cli/util/Config.js [hunk 2]
```diff
@@ -42,6 +42,12 @@ export type ConfigT = {
   getBlacklistRE(): RegExp,
 
   /**
+   * Specify any additional polyfill modules that should be processed
+   * before regular module loading.
+   */
+ getPolyfillModuleNames?: () => Array<string>,
```
Comment (jeanlauliac) [inline local-cli/util/Config.js]: See `defaultConfig` down that file.

**Comment by jeanlauliac** [comment]
To confirm: it's only a tiny change ^_^  Lots of the code in the rev I pointed out at is not relevant as it deals with the CLI args and the "default", but we don't want to add these, just the config with an empty function by default.

**Comment by eacaps** [comment]
Yeah, sorry github e-mail confused me so I saw that before these comments and assumed you were talking about something else. I think these changes are what you were asking for, I didn't get a chance to test them out quite yet but let me know if it looks good.

**Pull request closed without merging**